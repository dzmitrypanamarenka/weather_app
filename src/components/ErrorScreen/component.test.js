import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import { ErrorScreen } from '../';

describe('Snapshot test',() => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(<ErrorScreen />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Component render',() => {
  it('should shallow render ErrorScreen component', () => {
    const wrapper = shallow(<ErrorScreen />);
    expect(wrapper.find('.error-screen').length).toEqual(1);
  });
  it('should check mount component', () => {
    const wrapper = mount(<ErrorScreen />);
    const mountedComponent = wrapper.find('.text.-title');
    const text = 'Oooops :/ Something goes wrong!';
    expect(mountedComponent.length).toEqual(1);
    expect(mountedComponent.text()).toEqual(text);
  });
});
