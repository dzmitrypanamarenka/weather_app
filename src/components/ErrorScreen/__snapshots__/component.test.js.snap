// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test should match snapshot 1`] = `
<div
  className="error-screen"
>
  <span
    className="text -title"
  >
    Oooops :/ Something goes wrong!
  </span>
  <span
    className="text -message"
  />
</div>
`;
